restli-tools/src/main/resources/apiVmTemplates/rest.batch_delete.vm

```diff
@@ -0,0 +1,53 @@
+#*
+   Copyright (c) 2021 LinkedIn Corp.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*#
+  #doc($method.schema.doc)
+  public CompletionStage<Map<${spec.keyClassName}, UpdateStatus>> batchDelete(
+      Set<$spec.keyClassName> ids#if( $method.hasParams()),#end
+      #**##methodParams($method)##
+      ) {
+    Map<String, Object> queryParams = new HashMap<>($method.getQueryParamMapSize());
+    Map<String, Class<?>> queryParamClasses = #if($method.hasParams())new HashMap<>($method.getQueryParamMapSize());#else Collections.emptyMap();#end
+    #fillQueryParams($method)
+    queryParams.put(RestConstants.QUERY_BATCH_IDS_PARAM, ids);
+    BatchDeleteRequest<${spec.keyClassName}, ${spec.entityClassName}> request = new BatchDeleteRequest<>(
+        Collections.emptyMap(),
+        Collections.emptyList(),
+        queryParams,
+        queryParamClasses,
+        _resourceSpec,
+        ORIGINAL_RESOURCE_PATH,
+        ## TODO Path keys (sub-resources)
+        Collections.emptyMap(),
+        RestliRequestOptions.DEFAULT_OPTIONS);
+    CompletableFuture<Map<${spec.keyClassName}, UpdateStatus>> result = new CompletableFuture<>();
+    Task<Void> responseTask = _client.createTask(request).transform("Task to completion stage",
```

Comment:
Nit: I feel this completionStage logic is so common that we might be able to extract it and re-use.

The real difference is something like `(responseTry) -> responseTry.get().getEntity().getResults());`

Reply:
Agree, we can do it in a follow-up. I wanted to get the full picture of all APIs before we refactor it.